com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Artistic/Normal/NormalFromTextureNode.cs

```diff
@@ -0,0 +1,135 @@
+using System.Collections.Generic;
+using Usage = UnityEditor.ShaderGraph.GraphDelta.GraphType.Usage;
+
+namespace UnityEditor.ShaderGraph.Defs
+{
+
+    internal class NormalFromTextureNode : IStandardNode
+    {
+        static string Name = "NormalFromTexture";
+        static int Version = 1;
+        public static NodeDescriptor NodeDescriptor => new(
+            Version,
+            Name,
+            new FunctionDescriptor[] {
+                new(
+                    1,
+                    "3Samples",
+@"
+{
+    //3 sample version - only works on square textures with height in the red channel
+    //Offset = pow(Offset, 3) * 0.1;
+    //normalSample = SAMPLE_TEXTURE2D(Texture, Sampler, UV);
+    //va.z = ((SAMPLE_TEXTURE2D(Texture, Sampler, (UV.x + Offset, UV.y)).x) - normalSample) * Strength;
+    //vb.z = ((SAMPLE_TEXTURE2D(Texture, Sampler, (UV.x, UV.y + Offset)).x) - normalSample) * Strength;
+    //Out = normalize(cross(va.xyz, vb.xyz));
+    Out = float3(0,0,1);
+}",
+                    new ParameterDescriptor("Texture", TYPE.Vec4, Usage.In),//fix type
+                    new ParameterDescriptor("UV", TYPE.Vec2, Usage.In),//add default UVs
+                    new ParameterDescriptor("Sampler", TYPE.Vec2, Usage.In),//fix type
+                    new ParameterDescriptor("Offset", TYPE.Float, Usage.In),
```

Comment:
in sg1 `Offset `has a default value `0.5 `and `Strength `has a value of `8`

Reply:
Do you know how to remove files from the PR without deleting them locally?  I added the NormalFromTextureNode file by accident.  It's one of the texture sample nodes that's not ready yet.